clients/client/src/main/java/org/projectnessie/client/api/CreateReferenceBuilder.java

```diff
@@ -29,11 +31,81 @@
  * @since {@link NessieApiV1}
  */
 public interface CreateReferenceBuilder {
+  /**
+   * Sets the name of the reference that contains the hash of the reference-to-be-created.
+   *
+   * <p>This reference name will be used for authorization purposes to validate that the caller has
+   * read access to the provided hash.
+   *
+   * <p>If not explicitly set, the default branch will be used as the source reference name.
+   *
+   * <p>The name and hash of the reference to be created is set via {@link #reference(Reference)}.
+   *
+   * @param sourceRefName is the name of the reference that contains the hash of the
+   *     reference-to-be-created.
```

Comment:
`It must be present if {@link Reference#getHash()} is present in {@link #reference(Reference)}`

Reply:
I'd move this statement into the main javadoc body - easier to see for API users.

Reply:
I think the logic may actually be more complex... What if we set the hash but do not set the source ref? Will the server not check the hash to be reachable from the default branch?

Reply:
I don't think we have code that restricts this behaviour. 
So, It is not mandatory to provide `sourceRefName`. If not provided, it uses the default branch. 
So, this statement is no need add ?

cc: @dimas-b , @nastra

Reply:
Also  java doc of reference() covers this
> If the hash is specified, it should be on the {@link #sourceRefName(String)}